app/actions/dat-middleware.js

```diff
@@ -387,19 +387,29 @@ export default class DatMiddleware {
       await mkdirp(this.dataDir)
     } catch (_) {}
 
-    const [ datOpts, paused ] = await Promise.all([
+    const [datOpts, paused] = await Promise.all([
       readJSON(joinPath(this.dataDir, 'dats.json')),
       readJSON(joinPath(this.dataDir, 'paused.json'))
     ])
 
     for (const key of Object.keys(datOpts)) {
-      const opts = JSON.parse(datOpts[key])
-      this.internalAddDat({
-        key,
-        path: opts.dir,
-        paused: paused[key],
-        ...opts
-      })
+      let datdata
+
+      // It is for compatibility.
+      try {
+        datdata = JSON.parse(datOpts[key])
```

Comment:
Why did you decide to use a try-catch here? It is very slow; a `typeof` test would have been faster and it wouldn't catch other weird cases.